sdk/python/feast/image_utils.py

```diff
@@ -0,0 +1,279 @@
+            embeddings.append(normalized.flatten().tolist())
+
+        return embeddings
+
+
+def combine_embeddings(
+    text_embedding: List[float],
+    image_embedding: List[float],
+    strategy: str = "weighted_sum",
+    text_weight: float = 0.5,
+    image_weight: float = 0.5,
+) -> List[float]:
+    """
+    Combine text and image embeddings search.
+    This function provides several strategies for combining embeddings from
+    different modalities (text and image) into a single vector for search.
+
+    Args:
+        text_embedding: Text embedding vector
+        image_embedding: Image embedding vector
+        strategy: Combination strategy. Options:
+            - "weighted_sum": Weighted sum of aligned embeddings (default)
+            - "concatenate": Concatenate embeddings end-to-end
+            - "average": Simple average of aligned embeddings
+        text_weight: Weight for text embedding (for weighted strategies)
+        image_weight: Weight for image embedding (for weighted strategies)
+
+    Returns:
+        Combined embedding vector as list of floats
+
+    Raises:
+        ValueError: If strategy is invalid or weights don't sum to 1.0
+
+    Examples:
+        Weighted combination (emphasize image)::
+
+            combined = combine_embeddings(
+                [0.1, 0.2], [0.8, 0.9],  # text_emb, image_emb
+                strategy="weighted_sum",
+                text_weight=0.3, image_weight=0.7
+            )
+
+        Concatenation for full information::
+
+            combined = combine_embeddings(
+                [0.1, 0.2], [0.8, 0.9],  # text_emb, image_emb
+                strategy="concatenate"
+            )
+    """
+    if strategy == "weighted_sum":
+        if abs(text_weight + image_weight - 1.0) > 1e-6:
+            raise ValueError(
+                "text_weight + image_weight must equal 1.0 for weighted_sum"
+            )
+
+        max_dim = max(len(text_embedding), len(image_embedding))
+        text_padded = text_embedding + [0.0] * (max_dim - len(text_embedding))
+        image_padded = image_embedding + [0.0] * (max_dim - len(image_embedding))
+
+        combined = [
+            text_weight * t + image_weight * i
+            for t, i in zip(text_padded, image_padded)
+        ]
+        return combined
+
+    elif strategy == "concatenate":
+        return text_embedding + image_embedding
+
+    elif strategy == "average":
+        max_dim = max(len(text_embedding), len(image_embedding))
+        text_padded = text_embedding + [0.0] * (max_dim - len(text_embedding))
+        image_padded = image_embedding + [0.0] * (max_dim - len(image_embedding))
+
+        combined = [(t + i) / 2.0 for t, i in zip(text_padded, image_padded)]
+        return combined
+
+    else:
+        raise ValueError(
+            f"Unknown combination strategy: {strategy}. "
+            f"Supported strategies: weighted_sum, concatenate, average"
```

Comment:
small nit, technically you should just make the strategy an enum so the valueerror can be maintained easier